### PR TITLE
Particle Evaluating Paxel Expressions

### PR DESCRIPTION
**Pull request by piotrswigon**

### CODE REVIEW

File: java/arcs/core/data/expression/EvaluatorParticle.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                is CollectionType<*> -> {
+                    val collectionHandle = handle as WriteCollectionHandle<EntityBase>
+                    (result as Sequence<MapScope<*>>).forEach {
+                        collectionHandle.store(entityFromScope(it, schema))
+                    }
+                }
+            }
+        }
+    }
+
```
Comment (cromwellian) [inline java/arcs/core/data/expression/EvaluatorParticle.kt:106]: TODO: eventually Scope.lookup implementation will be able to delegate to EntityBase directly and avoid a copy

File: java/arcs/core/data/expression/EvaluatorParticle.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    private fun entityAsScope(entity: EntityBase, schema: Schema): MapScope<*> =
+        // TODO: Support collection fields.
+        schema.fields.singletons.mapValues { (name, _) ->
+            entity.getSingletonValue(name)
+        }.asScope()
+
```
Comment (cromwellian) [inline java/arcs/core/data/expression/EvaluatorParticle.kt:112]: Eventually we will replace this by supplying a ScopeBuilder to the Evaluator that can set the fields directly without an intermediate map. Not important now, but will be when we have a lot of data and iterating over largish collections

File: java/arcs/core/data/expression/EvaluatorParticle.kt (hunk 1 above)
Comment (piotrswigon) [reply java/arcs/core/data/expression/EvaluatorParticle.kt:106]: Added TODO

File: java/arcs/core/data/expression/EvaluatorParticle.kt (hunk 2 above)
Comment (piotrswigon) [reply java/arcs/core/data/expression/EvaluatorParticle.kt:112]: Added TODO

**Pull request merged**